crates/router/src/connector/bluesnap.rs

```diff
@@ -1075,8 +1030,34 @@ impl api::IncomingWebhook for Bluesnap {
             serde_urlencoded::from_bytes(request.body)
                 .into_report()
                 .change_context(errors::ConnectorError::WebhookResourceObjectNotFound)?;
+
+        let (card_transaction_type, processing_status) = match details.transaction_type {
+            bluesnap::BluesnapWebhookEvents::Decline
+            | bluesnap::BluesnapWebhookEvents::CcChargeFailed => Ok((
+                bluesnap::BluesnapTxnType::Capture,
+                bluesnap::BluesnapProcessingStatus::Fail,
+            )),
+            bluesnap::BluesnapWebhookEvents::Charge => Ok((
+                bluesnap::BluesnapTxnType::Capture,
+                bluesnap::BluesnapProcessingStatus::Success,
+            )),
+            bluesnap::BluesnapWebhookEvents::Unknown => {
+                Err(errors::ConnectorError::WebhookEventTypeNotFound)
+            }
+        }?;
+
+        let psync_struct = bluesnap::BluesnapPaymentsResponse {
+            processing_info: bluesnap::ProcessingInfoResponse {
+                processing_status,
+                authorization_code: None,
+                network_transaction_id: None,
+            },
+            transaction_id: details.reference_number,
+            card_transaction_type,
+        };
```

Comment:
Move this to from implementation based on event type Payments or Refunds.

Reply:
Resolving As per discussion